auth/standard_role_manager.cc

```diff
@@ -248,9 +248,13 @@ future<> standard_role_manager::start() {
             }
             return make_ready_future<>();
         }).then([this] {
-            _stopped = auth::do_after_system_ready(_as, [this] {
-                return seastar::async([this] {
-                    if (legacy_mode(_qp)) {
+            auto work_done = seastar::make_lw_shared<seastar::promise<>>();
```

Comment:
so you spawn async only to block it later? unless it's legacy because then you don't want to block

wouldn't  doing just:
if legacy(...) {
return seastar::async(...) /*current master's code here*/
}
return  create_default_role_if_missing(...);

have the same effect?

Reply:
btw perhaps corotinizing (in separate commit) would make it easier to understand the flow

Reply:
what do you mean? there is no `auto work_done` on master